### PR TITLE
Throw copy exception on invalid utf8 string

### LINKED ISSUE #2148: Upper and Lower do not handle invalid UTF-8
If we call `Upper()` or `Lower()` with binary data (anything non-UTF-8), our application simply hangs.

This is because we don't check the return value of `utf8proc_codepoint`, which is `-1` when UTF-8 is invalid. Our size goes unmodified, and we loop forever. Worse, the assert we have simply does nothing, since (as far as I can tell), it doesn't check that the codepoint is non-negative.

Reproduce with:

`LOAD FROM "test.csv" RETURN upper(column0);`

Where you can create test.csv by running:

`python -c "open('test.csv', 'wb').write(b'\xff')"`

### PR DESCRIPTION
**Pull request by acquamarin**
Closes #2148

### CODE REVIEW

File: src/processor/operator/persistent/reader/csv/driver.cpp [hunk 1]
```diff
@@ -340,6 +341,9 @@ void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view
             vector, rowToAdd, reinterpret_cast<char*>(blobBuffer.get()), blobLen);
     } break;
     case LogicalTypeID::STRING: {
+        if (!utf8proc::Utf8Proc::isValid(strVal.data(), strVal.length())) {
+            throw common::CopyException{"Invalid UTF8-encoded string."};
```
Comment (ray6080) [inline src/processor/operator/persistent/reader/csv/driver.cpp:345]: Can we also print the strVal in error message?

**Review by ray6080 — APPROVED** [review]

**Comment by Riolku** [comment]
If you add `Closes #2148` to your commit body, you needn't manually close the linked issue on merge.

**Comment by Riolku** [comment]
I'm fairly certain CSV reading isn't the only place invalid UTF-8 can enter a string column. We need to make sure that it every cast function fails on invalid UTF-8.

This is yet another reason to create a harmonized type system (besides better support across all parts of the system).

File: src/processor/operator/persistent/reader/csv/driver.cpp (hunk 1 above)
Comment (acquamarin) [reply src/processor/operator/persistent/reader/csv/driver.cpp:345]: I think printing the invalid utf8 characters is undefined behaviour and causes problems when writing the exception test for utf8 strings.

**Comment by Riolku** [comment]
See #2148 for why this PR is not sufficient. We should probably aim to fix this before the next major release, it's not a small task in my opinion.

**Pull request merged**